### ISSUE TITLE
Question about rank-0 tensors + Vector dialect

### ISSUE THREAD

**Issue by banach-space** [onnx/onnx-mlir#3029]
Hi folks,

I'm new to **onnx-mlir**, so please bear with me.

Does **onnx-mlir** have an opinion on rank-0 "tensors"? We've been discussing 0-D vectors in the [Vector](https://mlir.llvm.org/docs/Dialects/Vector/) dialect (used by Triton-CPU), and I want to ensure we’re not overlooking any critical use cases:

* https://discourse.llvm.org/t/rfc-should-we-restrict-the-usage-of-0-d-vectors-in-the-vector-dialect

Any input would be greatly appreciated (or even better, feedback directly in the Discourse thread. :blush:). Pointers on the usage of `Vector` dialect within **onnx-mlir** would be particular helpful.

Thank you!

-Andrzej

**Comment by AlexandreEichenberger** [comment]
@banach-space if anything, the confusion between scalar, 0-D vectors, or 1D vector with 1 element is tiresome :-)

**Comment by banach-space** [comment]
Agreed! We’re definitely keen to bring some clarity and order to `Vector`. :)
